docs: added Palette CLI subcommand (#1421)
spectrocloud/librarium: content/docs/04-clusters/02-data-center/03-vmware.md

Reviewer: I noticed this in the doc preview but couldn't find it in any of the files - maybe VMware doc?  
![CleanShot 2023-07-18 at 15 10 26](https://github.com/spectrocloud/librarium/assets/117382432/04285130-52c2-4305-bcc7-d46b34a3084a)

---

# Create VMware Private Cloud Gateway (PCG)

> There are two supported PCG installation methods for VMware vSphere. You can use the Palette CLI, or you can use an OVA/OVF template. Review the prerequisites for each option to help you identify the correct install method.

Reviewer: ```suggestion
There are two supported PCG installation methods for VMware vSphere. You can use the Palette CLI, or you can use an OVA/OVF template. Review the prerequisites for each option to help you identify the correct installation method.
```

<InfoBox>

A PCG cluster installation automatically creates a cloud account using the credentials entered at the time the PCG cluster is deployed. You can use this account to provision clusters across all tenant projects.

</InfoBox>

</Tabs.TabPane>

</Tabs>

<br />

---

# Upgrade PCG

Palette maintains the OS image and all configurations for the PCG. Periodically, the OS images, configurations, or other components need to be upgraded to resolve security or functionality issues. Palette releases such upgrades when required and communication about the same is presented in the form of an upgrade notification on the PCG.

Administrators should review the changes and apply them at a suitable time. Upgrading a PCG does not result in any downtime for the Tenant Clusters. During the upgrade process, the provisioning of new clusters might be temporarily unavailable. New cluster requests are queued while the PCG is being upgraded and are processed as soon as the PCG upgrade is complete.

<br />

### Delete PCG

Use the following steps to delete the PCG:

1. As a Tenant Administrator, navigate to the **Private Cloud Gateway** page under **Settings**.

> 2. Invoke the **Delete** action on the PCG instance that needs to be deleted.

Reviewer: ```suggestion
2. Invoke the **Delete** action on the PCG instance you want to delete.
```

> 3. The system performs a validation to ensure there are no running tenant clusters associated with the PCG instance being deleted. If such instances are found, the system presents an error. Delete relevant running tenant clusters and retry the deletion of the PCG.

Reviewer: ```suggestion
3. The system performs a validation to ensure there are no running tenant clusters associated with the PCG instance being deleted. If such instances are found, an error is displayed. Delete any running tenant clusters and retry deleting the PCG.
```

4. Delete the PCG Virtual Machines within vSphere.

<br />

### Resize PCG

> You can set up the PCG as a single-node or as a three-node cluster for high availability (HA). For production environments, we recommend three nodes. A PCG can be initially set up with one node and resized to three nodes later. Use the following steps to resize a single-node PCG cluster to a three-node PCG cluster.

Reviewer: ```suggestion
You can set up the PCG as a single-node cluster or as a three-node cluster for high availability (HA). For production environments, we recommend three nodes. A PCG can be initially set up with one node and resized to three nodes later. Use the following steps to resize a single-node PCG cluster to a three-node PCG cluster.
```

1. As a Tenant Administrator, navigate to the **Private Cloud Gateway** page under **Settings**.

2. Invoke the resize action for the relevant PCG instance.

3. Update the size from one (1) to three (3).

4. The PCG upgrade begins shortly after the update. Two new nodes are created on vSphere, and the PCG is upgraded to a three-node cluster.

<InfoBox>
Scaling a three-node cluster down to a single-node cluster is not permitted.<p></p> A load balancer instance is launched even for a single-node PCG to support future expansion.
</InfoBox>

# IP Address Management

> Palette supports DHCP as well as Static IP based allocation strategies for the VMs that are launched during cluster creation. IP Pools can be defined, using a range or a subnet. Administrators can define one or more IP pools linked to a PCG.

Reviewer: ```suggestion
Palette supports both DHCP and Static IP-based allocation strategies for the VMs that are launched during cluster creation. IP Pools can be defined using a range or a subnet. Administrators can define one or more IP pools linked to a PCG.
```

> Clusters created using a PCG can select from the IP pools linked to the corresponding PCG. By default, IP Pools are shared across multiple clusters, but can optionally be restricted to a cluster.

Reviewer: ```suggestion
Clusters created using a PCG can select from the IP pools linked to the corresponding PCG. By default, IP Pools are shared across multiple clusters but can optionally be restricted to a cluster.
```
